### DISCUSSION TITLE
Test all of the values in a map with mql

### DISCUSSION THREAD

**Discussion by scottford-io** [mondoohq/cnquery#678] [Q&A]
I am trying to test the arguments of a `terraform.module` whose value is a map of key-value pairs. The example I have only has one key called `module` defined, but the way the module is written, you can have as many keys as you want. 

I know I can write at query that explicitly tests the key that defined `block.arguments['bucket_policy_only']['module'] != false` but the key can be any string. 

How do I test all of the values that are defined? 🤔 

```
      arguments: {
        bucket_policy_only: {
          module: false
        }
        location: "var.location"
        names: [
          0: "module"
        ]
        prefix: "uniform-bucket-level-access-fail-"
        project_id: "var.project_id"
        source: "terraform-google-modules/cloud-storage/google"
      }
```

```
cnquery> terraform.modules.where( source == "registry.terraform.io/terraform-google-modules/cloud-storage/google" ) { block.arguments['bucket_policy_only']}
terraform.modules.where: [
  0: {
    block.arguments[bucket_policy_only]: {
      module: false
    }
  }
]
```

**Comment by chris-rock** [comment]
Can you share the HCL snippet as an example? I am not 100% sure I understand what you want to test for.

**Comment by scottford-io** [comment]
@chris-rock 

```hcl
module "uniform_bucket_level_access_fail" {
  source  = "terraform-google-modules/cloud-storage/google"

  prefix       = "uniform-bucket-level-access-fail-"
  names = [ "module" ]
  project_id = var.project_id
  location   = var.location

  bucket_policy_only = { 
    module = false 
  }

}
```

**Comment by arlimus** [comment]
We have a really nice solution for this with properly structured maps, but it turns out the testing for dict's (i.e. parsed JSON or in this case HCL structures) is quite restricting. You can still run the test via accessing all the `.values()` but it doesn't return the item that tests badly because `values` removes the keys that are responsible for the failure. I'll get that fixed :)

**Reply by scottford-io** [reply]
@arlimus with `.values()` should I use `_ != false`?
